src/components/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from "react";
+
+export type CheckBoxProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  "onChange"
+> & {
+  onChange: (value: boolean) => void;
+};
+
+const CheckBox: React.FC<CheckBoxProps> = ({
+  disabled,
+  onChange,
+  ...props
+}) => {
+  const [state, setState] = useState(false);
+  const handleClick = () => {
+    setState((state) => !state);
+    onChange(state);
+  };
```

Comment:
сложная логика, тут не должно состояние хранится внутри компонента, оно и так хранится за ним, надо просто
```suggestion
  const handleClick = () => {
    onChange(!checked);
  };
```

+ для всех функций надо подумать надо тем, чтобы обернуть их в `useCallback`. https://ru.reactjs.org/docs/hooks-reference.html